lib/audited/auditor.rb

```diff
@@ -271,9 +264,7 @@ def audited_columns
 
       # We have to calculate this here since column_names may not be available when `audited` is called
       def non_audited_columns
-        @non_audited_columns ||= audited_options[:only].present? ?
-                                 column_names - audited_options[:only] :
-                                 default_ignored_attributes | audited_options[:except]
+        @non_audited_columns ||= calculate_non_audited_columns
```

Comment:
For here and also `audited_columns`, I am not sure that we gain anything from memoizing these values. And as I mentioned in my commit message, taking away the memoization breaks a bunch of stuff which I think may hint at the existence of another problem. Not sure though.